### ISSUE TITLE
parseListOfPoints sometimes has empty "nums"

### ISSUE THREAD

**Issue by launchpad-issue-importer** [scour-project/scour#87]
I get the following traceback

```
scour.py -i lib/images/cover-chapter-2.svg -o build/lib/images/cover-chapter-2.svg --remove-metadata --quiet
Traceback (most recent call last):
  File "/home/sebastien/Local/share/scour/scour.py", line 3183, in <module>
    out_string = scourString(in_string, options).encode("UTF-8")
  File "/home/sebastien/Local/share/scour/scour.py", line 2952, in scourString
    cleanPolygon(polygon, options)
  File "/home/sebastien/Local/share/scour/scour.py", line 2164, in cleanPolygon
    pts = parseListOfPoints(elem.getAttribute('points'))
  File "/home/sebastien/Local/share/scour/scour.py", line 2136, in parseListOfPoints
    prev = nums[len(nums)-1]
IndexError: list index out of range
```

and the following fixes it

```
diff scour-fixed.py scour.py
2136,2141c2136,2138
< 					if len(nums) > 0:
< 						prev = nums[len(nums)-1]
< 						if prev[len(prev)-1] in ['e', 'E']:
< 							nums[len(nums)-1] = prev + '-' + negcoords[j]
< 						else:
< 							nums.append( '-'+negcoords[j] )
---
> 					prev = nums[len(nums)-1]
> 					if prev[len(prev)-1] in ['e', 'E']:
> 						nums[len(nums)-1] = prev + '-' + negcoords[j]
```

------------------------------------
Imported from Launchpad using lp2gh.

 * date created: 2012-11-06T15:27:47Z
 * owner: sebastien-type-z
 * the launchpad url was https://bugs.launchpad.net/bugs/1075592

**Comment by launchpad-issue-importer** [comment]
(by sebastien-type-z)

**Comment by launchpad-issue-importer** [comment]
(by louis-simard-deactivatedaccount)
cynthia@alakazam ~/scour/scour$ ./scour.py -i ~/temp/cover-chapter-2.svg -o ~/temp/cover-chapter-2.opt.svg --remove-metadata --quiet
cynthia@alakazam ~/scour/scour$ ./scour.py --version
0.26
cynthia@alakazam ~/scour/scour$ ls -la ~/temp/cover*
-rw-rw-r-- 1 cynthia cynthia 26918 Nov  6 13:55 /home/cynthia/temp/cover-chapter-2.opt.svg
-rw-rw-r-- 1 cynthia cynthia 32477 Nov  6 13:53 /home/cynthia/temp/cover-chapter-2.svg

I'm not positive this bug still occurs. What version of Scour are you running, and is it the latest code in the Bazaar repository? (See Code, above in Launchpad, to know how to get the code freom Bazaar.)

**Comment by launchpad-issue-importer** [comment]
(by u-chris-k)
I'm using 0.26 and see this error, as well.

$ python ~/Applications/scour/scour.py -i oak-color.svg -o output.svg
scour 0.26
Copyright Jeff Schiller, Louis Simard, 2010
Traceback (most recent call last):
  File "/Users/chrish/Applications/scour/scour.py", line 3183, in <module>
    out_string = scourString(in_string, options).encode("UTF-8")
  File "/Users/chrish/Applications/scour/scour.py", line 2952, in scourString
    cleanPolygon(polygon, options)
  File "/Users/chrish/Applications/scour/scour.py", line 2164, in cleanPolygon
    pts = parseListOfPoints(elem.getAttribute('points'))
  File "/Users/chrish/Applications/scour/scour.py", line 2136, in parseListOfPoints
    prev = nums[len(nums)-1]
IndexError: list index out of range

**Comment by launchpad-issue-importer** [comment]
(by a13b)
I also see this bug with two files, one of them being:
http://upload.wikimedia.org/wikipedia/commons/c/c2/Dublin_Rail_Network3.svg

The "s" parameter received by parseListOfPoints(s) is:
-7.227,-7.227 7.227,-7.227 7.227,7.227 -7.227,7.227

The parseListOfPoints method crashes if the first number is negative. That method needs some cleanup.

**Comment by Ede123** [comment]
This was fixed in Scour 0.32, see ab1aa0e2f8cb3faf1487249eea56cf83906b1f39
